core/network/impl/router_libp2p.cpp

```diff
@@ -155,37 +179,68 @@ namespace kagome::network {
         });
   }
 
-  void RouterLibp2p::handleTransactionsProtocol(
+  void RouterLibp2p::handleBlockAnnouncesProtocol(
       std::shared_ptr<Stream> stream) const {
-    auto rw = std::make_shared<libp2p::basic::MessageReadWriterUvarint>(stream);
-    rw->read([wself{weak_from_this()}, stream, rw](auto read_result) {
-      auto self = wself.lock();
-      if (!self) return;
+    Status status_msg;
 
-      if (!read_result) {
-        self->log_->error("Error while reading handshake: {}",
-                          read_result.error().message());
-        return stream->reset();
-      }
+    /// TODO(iceseer): use last finalized and best number
```

Comment:
When should this TODO be implemented?

Reply:
The next issue https://github.com/soramitsu/kagome/issues/589

Reply:
Please reflect it in the comment.

Reply:
done